### ISSUE TITLE
Feature request: `DataParallel` like in PyTorch

### ISSUE THREAD

**Issue by ngphuoc** [denizyuret/Knet.jl#603]
This supports multiple GPUs training by automatically splitting data along the batch dimension and sending the splits to multiple GPUs in the forward pass. In the backward pass, gradients from all GPUs are summed into the main thread.

Reference: https://pytorch.org/docs/master/generated/torch.nn.DataParallel.html
